code/core/src/phone/java/com/adobe/marketing/mobile/services/ui/notification/PushTemplateImageUtil.kt

```diff
@@ -83,10 +53,105 @@ internal object PushTemplateImageUtil {
      *
      * @param cacheService the AEPSDK [CacheService] to use for caching or retrieving
      * downloaded image assets
-     * @param uri [String] containing an image asset url
+     * @param uriList [String] containing an image asset url
      * @return [Bitmap] containing the image referenced by the `String` uri
      */
-    internal fun downloadImage(cacheService: CacheService, uri: String?): Bitmap? {
+    internal fun downloadImage(
+        cacheService: CacheService,
+        uriList: List<String?>
+    ): Int {
```

Comment:
Returning only count of downloaded images here to avoid storing a list of bitmaps in memory. The images can be fetched if/when needed using `getImageFromCache`